xcp_d/utils/hcp2fmriprep.py

```diff
@@ -4,165 +4,305 @@
+        "R.midthickness": "hemi-R_desc-hcp_midthickness",
+        "L.midthickness": "hemi-L_desc-hcp_midthickness",
+        "R.inflated": "hemi-R_desc-hcp_inflated",
+        "L.inflated": "hemi-L_desc-hcp_inflated",
+        "R.very_inflated": "hemi-R_desc-hcp_vinflated",
+        "L.very_inflated": "hemi-L_desc-hcp_vinflated",
+        "R.pial": "hemi-R_pial",
+        "L.pial": "hemi-L_pial",
+        "R.white": "hemi-R_smoothwm",
+        "L.white": "hemi-L_smoothwm",
+    }
+    for in_str, out_str in SURFACE_DICT.items():
+        surf_orig = os.path.join(
+            fsaverage_dir_orig,
+            f"{sub_id}.{in_str}.32k_fs_LR.surf.gii",
+        )
+        surf_fmriprep = os.path.join(
+            anat_dir_fmriprep,
+            f"{sub_ent}_space-fsLR_den-32k_{out_str}.surf.gii",
+        )
+        copy_dictionary[surf_orig] = [surf_fmriprep]
+
+    print("finished collecting anat files")
+
+    # Collect functional files to copy
+    subject_task_folders = sorted(glob.glob(os.path.join(in_dir, "*", "Results", "*")))
+    subject_task_folders = [
+        task for task in subject_task_folders if task.endswith("RL") or task.endswith("LR")
     ]
+    for subject_task_folder in subject_task_folders:
+        # NOTE: What is the first element in the folder name?
+        _, task_id, dir_id = os.path.basename(subject_task_folder).split("_")
+        task_ent = f"task-{task_id}"
+        dir_ent = f"dir-{dir_id}"
+        # TODO: Rename variable
+        run_foldername = os.path.basename(subject_task_folder)
+
+        # Find original task files
+        brainmask_orig_temp = os.path.join(subject_task_folder, "brainmask_fs.2.nii.gz")
+
+        bold_nifti_orig = os.path.join(subject_task_folder, f"{run_foldername}.nii.gz")
+        bold_nifti_fmriprep = os.path.join(
+            func_dir_fmriprep,
+            f"{sub_ent}_{task_ent}_{dir_ent}_{volspace_ent}_{res_ent}_desc-preproc_bold.nii.gz",
+        )
+        copy_dictionary[bold_nifti_orig] = [bold_nifti_fmriprep]
 
-    for i, j in zip(hcpfiles, newanatfiles):
-        copyfileobj_example(i, j)
-    print("finished converting anat files")
-    # get the task files
+        boldref_orig = os.path.join(subject_task_folder, "SBRef_dc.nii.gz")
+        boldref_fmriprep = os.path.join(
+            func_dir_fmriprep,
+            f"{sub_ent}_{task_ent}_{dir_ent}_{volspace_ent}_{res_ent}_boldref.nii.gz",
+        )
+        copy_dictionary[boldref_orig] = [boldref_fmriprep]
+
+        bold_cifti_orig = os.path.join(
+            subject_task_folder,
+            f"{run_foldername}_Atlas_MSMAll.dtseries.nii",
+        )
+        bold_cifti_fmriprep = os.path.join(
+            func_dir_fmriprep,
+            f"{sub_ent}_{task_ent}_{dir_ent}_space-fsLR_den-91k_bold.dtseries.nii",
+        )
+        copy_dictionary[bold_cifti_orig] = [bold_cifti_fmriprep]
+
+        # Grab transforms
+
+        native_to_t1w_orig = pkgrf("xcp_d", "/data/transform/itkIdentityTranform.txt")
```

Comment:
Maybe just use this for the template_to_t1w and t1w_to_template transforms as well, rather than the fake niftis.